sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/MetricsHelper.java

```diff
@@ -0,0 +1,187 @@
+    private static final String PARTITION_ID_KEY = "partitionId";
+    private static final String CONSUMER_GROUP_KEY = "consumerGroup";
+    private static final String STATUS_KEY = "status";
+
+    // since checkpoint store is stateless it might be used for endless number of eventhubs.
+    // we'll have as many subscriptions as there are combinations of fqdn, eventhub name, partitionId and consumer group.
+    // In the unlikely case it's shared across a lot of EH client instances, metrics would be too costly
+    // and unhelpful. So, let's just set a hard limit on number of subscriptions.
+    private static final int MAX_ATTRIBUTES_SETS = 100;
+
+    private static final String PROPERTIES_FILE = "azure-messaging-eventhubs-checkpointstore-blob.properties";
+    private static final String NAME_KEY = "name";
+    private static final String VERSION_KEY = "version";
+    private static final String LIBRARY_NAME;
+    private static final String LIBRARY_VERSION;
+    private static MeterProvider meterProvider = MeterProvider.getDefaultProvider();
+
+    private static final String UNKNOWN = "UNKNOWN";
+
+    static {
+        final Map<String, String> properties = CoreUtils.getProperties(PROPERTIES_FILE);
+        LIBRARY_NAME = properties.getOrDefault(NAME_KEY, UNKNOWN);
+        LIBRARY_VERSION = properties.getOrDefault(VERSION_KEY, UNKNOWN);
+    }
+
+    private final ConcurrentHashMap<String, TelemetryAttributes> common = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, TelemetryAttributes> checkpointFailure = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, TelemetryAttributes> checkpointSuccess = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, CurrentValue> seqNoSubscriptions = new ConcurrentHashMap<>();
+
+    private volatile boolean maxCapacityReached = false;
+
+    private final Meter meter;
+    private final LongGauge lastSequenceNumber;
+    private final LongCounter checkpointCounter;
+    private final boolean isEnabled;
+    MetricsHelper(MetricsOptions metricsOptions) {
+        if (metricsOptions != null && !metricsOptions.isEnabled()) {
+            this.meter = null;
+        } else {
+            this.meter = meterProvider.createMeter(LIBRARY_NAME, LIBRARY_VERSION, metricsOptions);
+        }
+        this.isEnabled = meter != null && meter.isEnabled();
+        if (isEnabled) {
+            this.lastSequenceNumber = this.meter.createLongGauge("messaging.eventhubs.checkpoint.sequence_number", "Last successfully checkpointed sequence number.", "seqNo");
+            this.checkpointCounter = this.meter.createLongCounter("messaging.eventhubs.checkpoints", "Number of checkpoints.", null);
+        } else {
+            this.lastSequenceNumber = null;
+            this.checkpointCounter = null;
+        }
+    }
+
+    /**
+     * Sets test meter provider.
+     */
+    static synchronized void setMeterProvider(MeterProvider testMeterProvider) {
```

Comment:
instead of creating this package-private method. Could we pass it in as a constructor parameter since this is something the metrics helper depends on?

```
MetricsHelper(MetricsOptions, MeterProvider) {
   this.metricsProvider = Objects.requreNotNull(meterProvider);
   ...
}
...
```
Calling class
```
var metricsHelper = new MetricsHelper(options, MeterProvider.getDefaultProvider());
```